main/src/library/Iterator.flix

```diff
@@ -60,6 +60,35 @@ namespace Iterator {
         };
         Iterator(done, next)
 
+    ///
+    /// Returns an iterator that unfolds `f` until the resource at `x` is null.
+    ///
+    /// Assumes that `f` eventually exhausts the resource at `x`.
+    ///
+    /// Assumes `f` may throw a Java Throwable.
+    ///
+    pub def ofThrowable(f: Unit -> a & Impure, x: Ref[a]): Iterator[Result[a, String]] =
+        let done = () -> Object.isNull(deref x);
+        let next = () -> {
+            try {
+                Ok(f())
+            } catch {
+                case e: ##java.lang.Throwable =>
```

Comment:
everything is caught here, including bug! calls from flix code. Maybe it should be more specific somehow?

Reply:
Agreed. How about `ofException` and we change it to java.lang.RuntimeException